stripe/src/main/java/com/stripe/android/net/StripeApiHandler.java

```diff
@@ -356,39 +388,47 @@ private static Token requestToken(
                     null, 0);
         }
 
-        try {
-            StripeResponse response = getStripeResponse(method, url, params, options);
+        StripeResponse response = getStripeResponse(method, url, params, options);
 
-            int rCode = response.getResponseCode();
-            String rBody = response.getResponseBody();
+        int rCode = response.getResponseCode();
+        String rBody = response.getResponseBody();
 
-            String requestId = null;
-            Map<String, List<String>> headers = response.getResponseHeaders();
-            List<String> requestIdList = headers == null ? null : headers.get("Request-Id");
-            if (requestIdList != null && requestIdList.size() > 0) {
-                requestId = requestIdList.get(0);
-            }
+        String requestId = null;
+        Map<String, List<String>> headers = response.getResponseHeaders();
+        List<String> requestIdList = headers == null ? null : headers.get("Request-Id");
+        if (requestIdList != null && requestIdList.size() > 0) {
+            requestId = requestIdList.get(0);
+        }
+
+        if (rCode < 200 || rCode >= 300) {
+            handleAPIError(rBody, rCode, requestId);
+        }
 
-            if (rCode < 200 || rCode >= 300) {
-                handleAPIError(rBody, rCode, requestId);
+        if (allowedToSetTTL) {
+            if (originalDNSCacheTTL == null) {
+                // value unspecified by implementation
+                // DNS_CACHE_TTL_PROPERTY_NAME of -1 = cache forever
+                java.security.Security.setProperty(
+                        DNS_CACHE_TTL_PROPERTY_NAME, "-1");
+            } else {
+                java.security.Security.setProperty(
+                        DNS_CACHE_TTL_PROPERTY_NAME, originalDNSCacheTTL);
             }
+        }
+        return rBody;
+    }
 
-            // Note that the 'finally' block runs immediately prior to this return statement.
-            return TokenParser.parseToken(rBody);
-        } catch(JSONException jsonException) {
+    private static Token requestToken(
```

Comment:
hm, so i don't have context around how `requestToken` is used, but it seems odd that it accepts any HTTP method – I'm pretty sure you can only POST to `v1/tokens` with a publishable key.